data/src/test/java/com/microsoft/azure/kusto/data/ClientRequestPropertiesTest.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.kusto.data;
+
+import org.json.JSONException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import java.util.concurrent.TimeUnit;
+
+public class ClientRequestPropertiesTest {
+    @Test
+    @DisplayName("test set/get timeout")
+    void testSetGetTimeout() {
+        ClientRequestProperties props = new ClientRequestProperties();
+        Long expected = TimeUnit.MINUTES.toMillis(100);
+
+        // before setting value should be null
+        Assertions.assertEquals(null, props.getTimeout());
+
+        props.setTimeout(expected);
+        Assertions.assertEquals(expected, props.getTimeout());
+    }
+
+    @Test
+    @DisplayName("test set/get timeout")
+    void testToString() throws JSONException {
```

Comment:
I'd avoid using the test prefix - everything in this class are tests, and also have the "@test" notation...
maybe name it timeout_ToString

Reply:
The link is explaining display names, not overall naming convention. I remember us checking this in the past and reached the conclusion that it is redundant. In this case for JUnit it isn't required in the method names...
https://dzone.com/articles/7-popular-unit-test-naming
https://www.petrikainulainen.net/programming/testing/writing-clean-tests-naming-matters/

I don't really mind one way or the other, but we didn't do this until now.